data_structures/dsu_path_compression.cpp

```diff
@@ -0,0 +1,183 @@
+/**
+ * @file
+ * @brief [DSU (Disjoint sets)](https://en.wikipedia.org/wiki/Disjoint-set-data_structure)
+ * @details
+ * It is a very powerful data structure that keeps track of different
+ * clusters(sets) of elements, these sets are disjoint(doesnot have a common element).
+ * Disjoint sets uses cases : for finding connected components in a graph,
+ * used in Kruskal's algorithm for finding Minimum Spanning tree.
+ * Operations that can be performed:
+ * 1) UnionSet(i,j): add(element i and j to the set)
+ * 2) findSet(i): returns the representative of the set to which i belogngs to.
+ * 3) get_max(i),get_min(i) : returns the maximum and minimum 
+ * Below is the class-based approach which uses the heuristic of path compression.
+ * Using path compression in findSet(i),we are able to get to the representative of i
+ * in O(1) time.
+ * @author [AayushVyasKIIT](https://github.com/AayushVyasKIIT)
+ * @see dsu_union_rank.cpp
+ */
+
+#include <iostream> /// for IO operations
+#include <vector>   /// for std::vector
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+/**
+ * @brief Disjoint sets union data structure, class based representation.
+ * @param n number of elements
+ */
+class dsu{
+    private:
+        vector<int> p; ///<keeps track of the parent of ith element
+        vector<int> depth; ///<tracks the depth(rank) of i in the tree
+        vector<int> setSize;///<size of each chunk(set)
+        vector<int> maxElement;/// <maximum of each set to which i belongs to
+        vector<int> minElement;/// <minimum of each set to which i belongs to
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). Let me know if you need any help. 🙂

Reply:
Did it!